support/kwok/kwok.go

```diff
@@ -0,0 +1,224 @@
+	version     string
+	timeout     time.Duration
+	kubecfgFile string
+	client      kubernetes.Interface
+	rc          *rest.Config
+}
+
+var kwokVersion = "v0.2.0"
+
+var _ support.E2EClusterProvider = &Cluster{}
+
+func (k *Cluster) Create(args ...string) (string, error) {
+	log.V(4).InfoS("Creating kwok cluster", "cluster", k.name)
+	if err := k.findOrInstallKWOKCtl(); err != nil {
+		return "", err
+	}
+
+	if _, ok := k.clusterExists(k.name); ok {
+		log.V(4).Info("Skipping kwok Cluster.Create: cluster already created: ", k.name)
+		return k.getKubeconfig()
+	}
+	command := fmt.Sprintf(`kwokctl create cluster --name %s`, k.name)
+
+	if k.timeout.Seconds() != 0 {
+		command = fmt.Sprintf("%s --wait %s", command, k.timeout.String())
+	}
+	if len(args) > 0 {
+		command = fmt.Sprintf("%s %s", command, strings.Join(args, " "))
+	}
+
+	log.V(4).Info("Launching:", command)
+	p := utils.RunCommand(command)
+	if p.Err() != nil {
+		return "", fmt.Errorf("failed to create kwok cluster: %s : %s", p.Err(), p.Result())
+	}
+
+	clusters, ok := k.clusterExists(k.name)
+	if !ok {
+		return "", fmt.Errorf("kwok Cluster.Create: cluster %v still not in 'cluster list' after creation: %v", k.name, clusters)
+	}
+	log.V(4).Info("kwok cluster available: ", clusters)
+
+	return k.getKubeconfig()
+}
+
+func (k *Cluster) CreateWithConfig(configFile string) (string, error) {
+	return k.Create("--config", configFile)
+}
+
+func (k *Cluster) Destroy() error {
+	log.V(4).Info("Destroying kwok cluster ", k.name)
+	if err := k.findOrInstallKWOKCtl(); err != nil {
+		return err
+	}
+
+	p := utils.RunCommand(fmt.Sprintf(`kwokctl delete cluster --name %s`, k.name))
+	if p.Err() != nil {
+		return fmt.Errorf("kwok: delete cluster failed: %s: %s", p.Err(), p.Result())
+	}
+
+	log.V(4).Info("Removing kubeconfig file ", k.kubecfgFile)
+	if err := os.RemoveAll(k.kubecfgFile); err != nil {
+		return fmt.Errorf("kwok: remove kubefconfig failed: %w", err)
+	}
+	return nil
+}
+
+func (k *Cluster) ExportLogs(_ string) error {
+	// TODO: Currently kwokctl has a way to export logs but you need to do that individually for each
+	// components. There is no bulk export. Might be a good idea to open an enhancement request there
+	// get that sorted so that we can directly use that here or we have to implement some way to
+	// individually export logs from each component into files under the destination location
+	return nil
+}
+
+func (k *Cluster) GetKubeCtlContext() string {
+	return fmt.Sprintf("kwok-%s", k.name)
+}
+
+func (k *Cluster) GetKubeconfig() string {
+	return k.kubecfgFile
+}
+
+func (k *Cluster) KubernetesClient() (kubernetes.Interface, error) {
+	if k.client != nil {
+		return k.client, nil
+	}
+
+	cfg, err := conf.New(k.kubecfgFile)
+	if err != nil {
+		return nil, err
+	}
+	k.rc = cfg
+	cs, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+	k.client = cs
+	return k.client, nil
+}
+
+func (k *Cluster) LoadDockerImage(_ string) error {
```

Comment:
I am a bit concerned about returning nil on a noop method.
We should make sure this is documented as a noop and/or returned an error.

Reply:
@vladimirvivien  I have removed this from this PR for now. So depending on which one gets merged first out of this and #239, we can handle the kwok bit accordingly.